js/components/main.js

```diff
@@ -482,37 +483,46 @@ class Main extends ImmutableComponent {
     currentWindow.on('maximize', function () {
       windowActions.setMaximizeState(true)
     })
-
     currentWindow.on('unmaximize', function () {
       windowActions.setMaximizeState(false)
     })
-
+    // NOTE: setTimeout is used to avoid spamming events
+    let resizeTimeout = null
     currentWindow.on('resize', function (event) {
-      windowActions.saveSize(event.sender.getSize())
+      if (resizeTimeout) {
+        clearTimeout(resizeTimeout)
+      }
+      resizeTimeout = setTimeout(function () {
+        const size = event.sender.getSize()
+        // NOTE: the default window size is whatever the last window resize was
+        appActions.defaultWindowParamsChanged(size)
+        windowActions.saveSize(size)
+      }, 1000)
     })
-
     currentWindow.on('focus', function () {
       windowActions.onFocusChanged(true)
     })
-
     currentWindow.on('blur', function () {
+      appActions.windowBlurred(currentWindow.id)
       windowActions.onFocusChanged(false)
     })
-
+    // NOTE: setTimeout is used to avoid spamming events
     let moveTimeout = null
     currentWindow.on('move', function (event) {
       if (moveTimeout) {
```

Comment:
re: my comment below, I didn't see this at first, but I still think debounce would be better. Can `move` and `resize` call a common method?